### PR TITLE
upgrade to webpack 5

### LINKED ISSUE #195: Upgrade to webpack v5
webpack 5 has been stable for a while now and we should be able to upgrade without any problems.

### PR DESCRIPTION
**Pull request by thomaschaaf**
This fix also includes some other PRs work.

Breaking:
node10 was removed and instead of webpack4 we are now using webpack 5.

fixes #195

### CODE REVIEW

File: src/webpack.config.js [hunk 1]
```diff
@@ -288,17 +290,6 @@ function plugins() {
     plugins.push(new ForkTsCheckerWebpackPlugin(forkTsCheckerWebpackOptions));
   }
 
-  if (ENABLE_CACHING) {
```
Comment (thomaschaaf) [inline src/webpack.config.js:291]: not needed anymore as webpack5 has caching built in

File: tests/typescript/handler.ts [hunk 2]
```diff
@@ -1,10 +1,6 @@
 import { hi } from "@my-org/some-lib/index";
 import { merhaba } from "./tsx/import";
 
-export const typescript4Inject = <A extends []>(
```
Comment (thomaschaaf) [inline tests/typescript/handler.ts:4]: this caused an eslint error and I did not understand what it did as the function is never called

**Comment by thomaschaaf** [comment]
@jayair could you review this please :)

**Comment by jayair** [comment]
@thomaschaaf This is awesome. Thank you.

Are there any breaking changes in this? Do you think this needs a canary release to test or is it stable enough that we could release it live?

**Comment by joewestcott** [comment]
@thomaschaaf I think you should use npmv7 to work with the `package-lock.json` file in this repo, your PRs currently downgrade to v1.

**Comment by thomaschaaf** [comment]
@jayair no known breaking changes

@joewestcott I'm not using node 15 yet with which npm v7 will be published. I personally think it does not make sense to use npm v7 before node 16 is released. As most developers will not have it, which will result in the `package-lock.json` effectively being ignored.

**Comment by thomaschaaf** [comment]
@jayair can we get this packaged?

**Comment by jayair** [comment]
@thomaschaaf I just need a little bit of time to go through and test all of this. It's a slightly big change. I'll get to it next weekend.

**Comment by jayair** [comment]
@thomaschaaf So I had a look at all your PRs. Thanks for all the hard work!

I think it make sense that we put them all together cut a `5.0.0` beta release to test. It seems like they can all go together right?

https://github.com/AnomalyInnovations/serverless-bundle/pull/211
https://github.com/AnomalyInnovations/serverless-bundle/pull/214

**Comment by thomaschaaf** [comment]
Yey! Yes it all belongs together.

**Comment by jayair** [comment]
Yeah agreed. If you can merge it into one PR then I'll cut a beta release.

Thank you!

**Comment by thomaschaaf** [comment]
@jayair  ready to make a new Version 🙌

**Comment by jayair** [comment]
Thank you! Pushed a new canary release https://github.com/AnomalyInnovations/serverless-bundle/releases/tag/v5.0.0-beta.0

Starting a new thread to help test it and collect feedback.

https://github.com/AnomalyInnovations/serverless-bundle/issues/229

**Pull request merged**